ui/dashboards/src/context/VariableProvider/VariableProvider.tsx

```diff
@@ -400,48 +466,40 @@ export type ExternalVariableDefinition = {
 export interface VariableProviderProps {
   children: ReactNode;
   initialVariableDefinitions?: VariableDefinition[];
-  /**
-   * The external variables allow you to give to the provider some additional variables, not defined in the dashboard and static.
-   * It means that you won´t be able to update them from the dashboard itself, but you will see them appear and will be able
-   * to modify their runtime value as any other variable.
-   * If one of the external variable has the same name as a local one, it will be marked as overridden.
-   * You can define one list of variable definition by source and as many source as you want.
-   * The order of the sources is important as first one will take precedence on the following ones, in case they have same names.
-   */
   externalVariableDefinitions?: ExternalVariableDefinition[];
-  builtinVariables?: BuiltinVariableDefinition[];
+  builtinVariableDefinitions?: BuiltinVariableDefinition[];
 }
 
 export function VariableProvider({
   children,
   initialVariableDefinitions = [],
   externalVariableDefinitions = [],
-  builtinVariables = [],
+  builtinVariableDefinitions = [],
 }: VariableProviderProps) {
-  const [store] = useState(createVariableSrvStore({ initialVariableDefinitions, externalVariableDefinitions }));
+  const [store] = useState(createVariableDefinitionStore({ initialVariableDefinitions, externalVariableDefinitions }));
 
   return (
-    <VariableStoreContext.Provider value={store}>
```

Comment:
Maybe this file can be renamed VariableDefinitionProvider?

Reply:
Just the file, or the Provider that goes with it? IMHO better to keep named simply VariableProvider as this one is "user-facing" (part of the embed use case https://github.com/perses/perses/blob/main/docs/user-guides/embedding-panels.md)

Reply:
Mmh yeah nevermind. If we want in the end to merge the variable providers in one, this might be a good idea to not rename it. Can you put a TODO in the js doc of this ``VariableProvider`` relating this secret wish ? :D